app/src/main/java/com/lucasmontano/openweathermap/MainActivity.kt

```diff
@@ -2,11 +2,20 @@ package com.lucasmontano.openweathermap
 
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
+import androidx.navigation.fragment.NavHostFragment
 
 class MainActivity : AppCompatActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
     }
+
+    override fun onBackPressed() {
+        val fragment =
+            this.supportFragmentManager.findFragmentById(R.id.nav_host_fragment) as? NavHostFragment
+        val currentFragment = fragment?.childFragmentManager?.fragments?.get(0) as? IOnBackPressed
+
+        currentFragment?.onBackPressed()?.takeIf { !it }?.let { super.onBackPressed() }
```

Comment:
we could also use **.not()** as we don't have a complex IF condition to be done :)
`onBackPressed()?.not()?.let { ... }`